pkg/obfuscator/ip.go

```diff
@@ -36,26 +44,51 @@ func (o *ipObfuscator) replace(s string) string {
 	ipv4matches := o.ipv4pattern.FindAllString(output, -1)
 	for _, m := range ipv4matches {
 		if ip := net.ParseIP(m); ip != nil {
-			output = strings.ReplaceAll(output, m, obfuscatedStaticIPv4)
-			o.ReportReplacement(m, obfuscatedStaticIPv4)
+			var replacement string
+			switch o.replacementType {
+			case schema.ObfuscateReplacementTypeStatic:
+				replacement = obfuscatedStaticIPv4
+			case schema.ObfuscateReplacementTypeConsistent:
+				if replacement = o.HasReplacement(m); replacement == "" {
+					o.ipv4count++
+					replacement = fmt.Sprintf(consistentIPv4Template, o.ipv4count)
+				}
+			default:
+				replacement = m
```

Comment:
should this actually happen?

Reply:
It shouldn't but this guard condition prevents the obfuscated output from being blank.